src/Pages/Homepage.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Container, Row, Col } from "react-bootstrap";
+import Header from '../Frame/Header/Header';
+import FreeWeekRotation from '../Components/FreeWeekRotation/FreeWeekRotation';
+import SearchField from '../Components/SearchField/SearchField';
+import { fetchChampionList } from '../Modules/APIGateway';
+
+class Homepage extends React.Component {
+    state = {
+        champions: null,
+    }
+
+    async componentDidMount() {
+        // Get all champions from the server when app starts
+        let champion_list = await fetchChampionList();
+
+        this.setState({champions: champion_list});
+    }
+
+    render() {
+        let free_week_rotation = "";
+        if (this.state.champions) {
+            free_week_rotation = <FreeWeekRotation champions={this.state.champions} />
```

Comment:
It is really weird to store a component like that. We should use a isReady or isLoading prop, and if we don't have the data yet, render a spinner and not an empty string.